### ISSUE TITLE
Unable to use changes output file from `pkg_deb` in further rules

### ISSUE THREAD

**Issue by phlax** [bazelbuild/rules_pkg#477]
Previously i was able to tar deb and changes files created by `pkg_deb` using `pkg_tar`

I did this by manually listing expected outputs as srcs for the pkg_tar rule, eg:

```

debs = (
     "envoy.deb"
)

changes = (
    "envoy_%s.changes" % version,
)

pkg_tar(
       name = "base_debs",
       extension = "tar",
       package_dir = "deb",
       srcs = [":%s" % f for f in debs + changes],
 )
 
```

this no longer seems to work

if i build the deb/s correctly i can see the changes files, but they are no longer available to subsequent rules

**Comment by aiuto** [comment]
That would be a consequence of dynamically building the file name based on the dynamic deb file name.
One solution would be to have both files in the DefaultInfo returned by the rule, then you could wire it up as:
```
pkg_deb(name = "deb", ...)
pkg_tar(
   name = "both",
   srcs = [":deb"],
)
```
Your example was just bundling it up in a tarball. Do you have a more complex use case at the heart of this?

**Comment by phlax** [comment]
> Do you have a more complex use case at the heart of this?

kind of, yes.

the debs in the bundled tarballs are signed, and then tested in containers for actual distros

i was hoping that `$(locations :deb)` would work, but afaict it doesnt, but would be happy to be wrong

atm, i have a v hacky and non-hermetic workaround of following the `realpath` of the deb to find the sibling changes file

**Comment by aiuto** [comment]
On Tue, Nov 23, 2021 at 8:31 AM phlax ***@***.***> wrote:

> Do you have a more complex use case at the heart of this?
>
> kind of, yes.
>
> the debs in the bundled tarballs are signed, and then tested in containers
> for actual distros
>
> i was hoping that $(locations :deb) would work, but afaict it doesnt, but
> would be happy to be wrong
>
That's good to know.  I think if we added the change to DefaulInfo that
would work too.  That's why
the full use case is always important. What people are trying to accomplish
is more useful than just
knowing the narrow thing they are trying to do.

**Comment by nacl** [comment]
Possibly related: #396

**Comment by aiuto** [comment]
The fix is in #484.  It is one line of output, but the more interesting part is the little used technique of using a `filegroup` to use an implicit output like the changes file into another rule.
